src/security.c

```
@@ -1,1012 +1,491 @@
-
-    /* Check primary counter */
-    if (security_state.primary_counter.valid)
-    {
-        crypto_auth_hmacsha256_state hmac_state;
-        crypto_auth_hmacsha256_init(&hmac_state, security_state.device_secret, sizeof(security_state.device_secret));
-        crypto_auth_hmacsha256_update(&hmac_state, (const uint8_t *)&security_state.primary_counter.value, sizeof(uint64_t));
-        crypto_auth_hmacsha256_final(&hmac_state, expected_hash);
-
-        /* Use constant-time comparison for security-critical data */
-        primary_valid = sodium_memcmp(expected_hash, security_state.primary_counter.hash, 32) == 0;
+int security_load_identity_keys(uint8_t* out_public_key, uint8_t* out_private_key) {
```

Clang (14.0.6)

[warning] pointer parameter 'out_public_key' can be pointer to const

(readability-non-const-parameter)